### PR TITLE
doc(NgSwitch): update API doc

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/angular2/src/core/directives/ng_switch.ts [hunk 1]
```diff
@@ -15,9 +18,9 @@ export class SwitchView {
 }
 
 /**
- * The `NgSwitch` directive is used to conditionally swap DOM structure on your template based on a
- * scope expression.
- * Elements within `NgSwitch` but without `NgSwitchWhen` or `NgSwitchDefault` directives will be
+ * Conditionally display child templates when their expression match the switch expression.
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_switch.ts]: _expression match_  -->  _expressions match / expression matches_

File: modules/angular2/src/core/directives/ng_switch.ts [hunk 2]
```diff
@@ -15,9 +18,9 @@ export class SwitchView {
...
+ *
+ * Elements within `NgSwitch` but without `NgSwitchWhen` nor `NgSwitchDefault` directives will be
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_switch.ts]: <sub>I am not a native speaker, but is `nor` correct here ? It sounds "funny".</sub>

File: modules/angular2/src/core/directives/ng_switch.ts [hunk 3]
```diff
@@ -25,18 +28,49 @@ export class SwitchView {
...
+ *       <template [ng-switch-when]="0">0, increment again</template>
+ *       <template [ng-switch-when]="1">1, increment again</template>
+ *       <template [ng-switch-when]="2">2, stop incrementing</template>
+ *       <template ng-switch-default>&gt; 2, STOP!</template>
+ *     </p>
+ *   `,
+ *   directives: [NgSwitch, NgSwitchWhen, NgSwitchDefault]
+ * })
+ * export class App {
+ *   value: number = 'init';
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_switch.ts]: I am sure TS will not be happy about it. Why not initialize as `0` ?

File: modules/angular2/src/core/directives/ng_switch.ts [hunk 4]
```diff
@@ -125,28 +159,24 @@ export class NgSwitch {
...
 export class NgSwitchWhen {
   // `_WHEN_DEFAULT` is used as a marker for a not yet initialized value
   _value: any = _WHEN_DEFAULT;
   _view: SwitchView;
+  private _switch: NgSwitch;
 
   constructor(viewContainer: ViewContainerRef, templateRef: TemplateRef,
-              @Host() private _switch: NgSwitch) {
+              @Host() ngSwitch: NgSwitch) {
+    this._switch = ngSwitch;
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_switch.ts]: OOC, why these changes ? Didn't the idiomatice `private _switch: NgSwitch` work just as well ?

File: modules/angular2/src/core/directives/ng_switch.ts [hunk 5]
```diff
@@ -125,28 +159,24 @@ export class NgSwitch {
 }
 
 /**
- * Defines a case statement as an expression.
+ * Displays the content when the expression evaluates to the same as the enclosing `NgSwitch`
+ * expression.
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_switch.ts]: This sentence sounds somewhat confusing (imo). Maybe something more like:

_"Displays this elements content when the expression evaluates to the same value as the enclosing `NgSwitch`'s expression."_

<sub>(But again, non-native speaker :))</sub>

File: modules/angular2/src/core/directives/ng_switch.ts (hunk 4 above)
Comment (vicb) [inline modules/angular2/src/core/directives/ng_switch.ts]: Seems like decorator + visibility causes issues with TS an is not used in other parts of the code base. We might revisit this across the code base once the issue is fixed.

**Comment by vicb** [comment]
@gkalpak thanks for the review

**Comment by gkalpak** [comment]
@vicb, thx for writing docs :grinning:

**Comment by vicb** [comment]
> thx for writing docs

That's pure fun, you should try !

**Comment by IgorMinar** [comment]
vicb please rebase.. sorry about the delay

**Comment by vicb** [comment]
rebased

File: modules/angular2/src/common/directives/ng_switch.ts [hunk 6]
```diff
@@ -6,7 +6,10 @@ import {ListWrapper, Map} from 'angular2/src/facade/collection';
 
 const _WHEN_DEFAULT = CONST_EXPR(new Object());
 
-export class SwitchView {
+/**
+ * @private
```
Comment (IgorMinar) [inline modules/angular2/src/common/directives/ng_switch.ts]: what's the purpose of `@private`? we are not exporting the class, so it's private by default.

File: modules/angular2/src/common/directives/ng_switch.ts [hunk 7]
```diff
@@ -164,15 +193,10 @@ export class NgSwitchWhen {
...
+ * value.
  *
- * Default case statements are displayed when no `NgSwitchWhen` match the `ng-switch` value.
- *
- * Example:
- *
- * ```
- * <template ng-switch-default>...</template>
- * ```
+ * see {@link NgSwitch} for more details and example.
```
Comment (IgorMinar) [inline modules/angular2/src/common/directives/ng_switch.ts]: Capital `S` in `see1

**Comment by IgorMinar** [comment]
otherwise lgtm
